pavelib/assets.py

```diff
@@ -20,75 +20,174 @@
+        "css_destination_dir": "",  # destination where css files would be placed
+        "lookup_paths": [],  # list of directories to be passed as lookup paths for @import resolution.
+    }
+
+    if theme_dir is empty or None then return sass directories for the given system only. (i.e. lms or cms)
+
+    :param system: name if the system for which to compile sass e.g. 'lms', 'cms'
+    :param theme_dir: absolute path of theme for which to compile sass files.
+    """
+    if system not in SYSTEMS:
+        raise ValueError("'system' must be one of ({allowed_values})".format(allowed_values=', '.join(SYSTEMS.keys())))
+    system = SYSTEMS[system]
+
+    applicable_directories = list()
+
+    # add common sass directories
+    applicable_directories.append({
+        "sass_source_dir": path("common/static/sass"),
+        "css_destination_dir": path("common/static/css"),
+        "lookup_paths": [
+            path("common/static"),
+            path("common/static/sass"),
+        ],
+    })
+
+    if theme_dir:
+        # Add theme sass directories
+        applicable_directories.extend(
+            get_theme_sass_dirs(system, theme_dir)
+        )
+    else:
+        # add system sass directories
+        applicable_directories.extend(
+            get_system_sass_dirs(system)
+        )
+
     return applicable_directories
 
 
+def get_theme_sass_dirs(system, theme_dir):
+    """
+    Return list of sass dirs that need to be compiled for the given theme. Also add theme sass directories to
+    THEME_SASS_WATCHER_DIRS variables so that they could be added to sass watcher later.
+
+    :param system: name if the system for which to compile sass e.g. 'lms', 'cms'
+    :param theme_dir: absolute path of theme for which to compile sass files.
+    """
+    if system not in ('lms', 'cms'):
+        raise ValueError('"system" must either be "lms" or "cms"')
+
+    dirs = []
+
+    system_sass = path(system) / "static" / "sass"
+    sass = theme_dir / system / "static" / "sass"
+    css = theme_dir / system / "static" / "css"
+
+    if sass.isdir():
+        css.mkdir_p()
+
+        # first compile lms sass files and place css in theme dir
+        dirs.append({
+            "sass_source_dir": system_sass,
+            "css_destination_dir": css,
+            "lookup_paths": [
+                sass / "partials",
+                system_sass / "partials",
+                system_sass,
+            ],
+        })
+
+        # now compile theme sass files and override css files generated from lms
+        dirs.append({
+            "sass_source_dir": sass,
+            "css_destination_dir": css,
+            "lookup_paths": [
+                sass / "partials",
+                system_sass / "partials",
+                system_sass,
+            ],
+        })
+
+        # Add theme dirs to sass watcher directories
+        THEME_SASS_WATCHER_DIRS.extend([sass, sass / 'partials'])
```

Comment:
I find it disconcerting that a method called ``get_theme_sass_dirs`` has a side effect of updating the watcher directories. IMO get methods should have no side effects. It also seems that if you call get multiple times for the same theme, then they will be added multiple times to the watcher directories.